src/components/App.tsx

```diff
@@ -4387,6 +4392,85 @@ class App extends React.Component<AppProps, AppState> {
     return true;
   };
 
+  private handleCanvasZoomUsingCtrlAndSpace = (
+    event: React.PointerEvent<HTMLElement>,
+  ): boolean => {
+    if (
+      !(
+        gesture.pointers.size <= 1 &&
+        (event.button === POINTER_BUTTON.WHEEL ||
+          (event.button === POINTER_BUTTON.MAIN &&
+            isHoldingSpace &&
+            event.ctrlKey) ||
+          isHandToolActive(this.state) ||
+          this.state.viewModeEnabled)
+      ) ||
+      isTextElement(this.state.editingElement)
+    ) {
+      return false;
+    }
+    event.preventDefault();
+
+    let previousX = event.clientX;
+
+    const onPointerMove = withBatchedUpdatesThrottled((event: PointerEvent) => {
+      const delta = previousX - event.clientX;
+      const sign = Math.sign(delta);
+      const absDelta = Math.abs(delta);
+
+      if (delta < 0) {
+        setCursor(this.canvas, CURSOR_TYPE.ZOOM_IN);
+      } else {
+        setCursor(this.canvas, CURSOR_TYPE.ZOOM_OUT);
+      }
+
+      let newZoom = this.state.zoom.value - delta / 200;
+      newZoom +=
+        Math.log10(Math.max(1, this.state.zoom.value)) *
+        -sign *
+        Math.min(1, absDelta / 10);
+
+      previousX = event.clientX;
+
+      this.translateCanvas((state) => ({
+        ...getStateForZoom(
+          {
+            viewportX: this.lastViewportPosition.x,
+            viewportY: this.lastViewportPosition.y,
+            nextZoom: getNormalizedZoom(newZoom),
+          },
+          state,
+        ),
+        shouldCacheIgnoreZoom: true,
+      }));
+    });
+
+    const teardown = withBatchedUpdates(
+      (lastPointerUp = () => {
+        lastPointerUp = null;
+        if (!isHoldingSpace) {
+          setCursorForShape(this.canvas, this.state);
+        }
+        this.setState({
+          cursorButton: "up",
+        });
+        this.savePointer(event.clientX, event.clientY, "up");
+        window.removeEventListener(EVENT.POINTER_MOVE, onPointerMove);
+        window.removeEventListener(EVENT.POINTER_UP, teardown);
+        window.removeEventListener(EVENT.BLUR, teardown);
+        onPointerMove.flush();
+      }),
+    );
+    window.addEventListener(EVENT.BLUR, teardown);
+    window.addEventListener(EVENT.POINTER_MOVE, onPointerMove, {
+      passive: true,
+    });
+    window.addEventListener(EVENT.POINTER_UP, teardown);
+
+    this.resetShouldCacheIgnoreZoomDebounced();
```

Comment:
this should be called inside the event handler after you `translateCanvas`

Reply:
Fixed in the last commit.